### PR TITLE
feat: COAL

### PR DESCRIPTION
**Pull request by robot9706**
- Implement a method to fetch image labels using the V2 registry API
- Add validation based on image labels (COAL)
- Refactored registry validation

### CODE REVIEW

File: web/crux/src/app/image/image.dto.ts [hunk 1]
```diff
@@ -26,6 +55,9 @@ export class ImageDto {
 
   @Type(() => BasicRegistryDto)
   registry: BasicRegistryDto
+
+  @ValidateNested()
+  validation: ImageValidation
```
Comment (m8vago) [inline web/crux/src/app/image/image.dto.ts:60]: COAL's ruleset should not bleed into crux's API structure. At this point it should only exist as labels IMHO.

File: web/crux/src/domain/image.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+export const ENVIRONMENT_VALUE_TYPES = ['string', 'boolean', 'int'] as const
+export type EnvironmentValueType = (typeof ENVIRONMENT_VALUE_TYPES)[number]
+
+export type EnvironmentRule = {
+  type: EnvironmentValueType
+  required?: boolean
+  default?: string
+}
+
+export const parseDyrectorioEnvRules = (labels: Record<string, string>): Record<string, EnvironmentRule> => {
```
Comment (m8vago) [inline web/crux/src/domain/image.ts:12]: Please write a brief summary with some valid example rules for this function.

File: web/crux/src/app/registry/registry-clients/unchecked-api-client.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    throw new CruxBadRequestException({ message: 'Unchecked registries have no tags API' })
+  }
+
+  async labels(image: string, tag: string): Promise<Record<string, string>> {
+    if (this.url === '') {
+      return {}
+    }
+
+    const labelClient = new V2Labels(this.url)
+    return labelClient.fetchLabels(image, tag)
```
Comment (m8vago) [inline web/crux/src/app/registry/registry-clients/unchecked-api-client.ts:23]: Unchecked registries have an offline use-case, we simply shouldn't even try to parse COAL rules with them. They are unchecked.

**Review by m8vago — CHANGES_REQUESTED** [review]

**Review by m8vago — APPROVED** [review]

**Comment by m8vago** [comment]
Failed to fetch labels with gitlab. Needs further development.

**Review by m8vago — APPROVED** [review]

**Review by polaroi8d — APPROVED** [review]

**Pull request merged**